### ISSUE TITLE
Add ability to set random seed 

### ISSUE THREAD

**Issue by rmarkello** [murraylab/brainsmash#4]
Hey `brainsmash` developers! :wave:

## Description

In working a bit more closely with `brainsmash` one thing that I found myself really wanting was the ability to set a random seed when creating a new `Base` class instance. That is, currently when I re-run a script that uses `Base` my surrogate brain maps vary (rather considerably!). I can set the seed outside of the `Base` class (e.g., `np.random.seed(1234)`), but having the ability to set the seed within each class instance be nice for when you're juggling multiple instances around!

## Proposed solution

I managed to hack something together in the interim (very bad, would not recommend), but I think the easiest way to address this would be to add a new parameter to the `Base` class (something like `seed` or `random_state`) that would then be used for all of the calls to `np.random` throughout the class methods.

In `Base.__init__` you could then check whether a seed was provided (since `sklearn` is a dependency already you can use `sklearn.utils.validation.check_random_state`):

```python
self._rs = check_random_state(seed)
```

and throughout the `Base` methods instead of `np.random` you'd just use `self._rs`!

Happy to open a PR making these changes, if desired. Also happy to discuss the relative utility of this in more detail!

**Comment by jbburt** [comment]
@rmarkello another great idea! I completely agree. This functionality is now implemented in `Base` as well as `Sampled`
